### PR TITLE
Update to new clang complete syntax

### PR DESCRIPTION
**Pull request by topisani**
I don't know when the clang syntax changed, but the old command didn't work for me with at least `clang >= 3.8`

This simply patches `buildCommand` to use the new syntax

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
   },
   "dependencies": {
     "vscode-languageserver": "^2.6.2",
-    "when-switch": "^0.2.0"
+    "when-switch": "^0.2.0",
+    "path": "*",
```
Comment (kube) [inline server/package.json]: Why did you add `path` and `vscode-uri` dependencies?

File: server/src/completion.ts [hunk 2]
```diff
@@ -102,16 +104,16 @@ const completionList = (output: string): ICompletionItem[] =>
...
       'clang',
-      '-cc1',
-      ...userFlags,
       '-fsyntax-only',
-      languageId === 'c' ? '-xc' : '-xc++',
-      '-code-completion-macros',
-      '-code-completion-at',
-      `-:${position.line + 1}:${position.character + 1}`
+      ...userFlags,
+      `-I${path.dirname(Uri.parse(doc.uri).fsPath)}`,
```
Comment (kube) [inline server/src/completion.ts]: Why do you add to include folders the folder of the current file ?
I'm not sure of the interest here.

File: server/src/completion.ts [hunk 3]
```diff
@@ -152,12 +154,15 @@ export const getCompletion =
       const command = buildCommand(config.userFlags, {
         line: position.line,
         character: column
-      }, document.languageId)
+      }, document)
+      console.log(command)
```
Comment (kube) [inline server/src/completion.ts]: Could you please remove these three unnecessary `console.log`?
(Including the one in `server/src/server.ts`)

**Review by kube — CHANGES_REQUESTED** [review]
Thanks for the syntax update.
Please just answer (and fix) remarks in my review before merging.

File: server/package.json (hunk 1 above)
Comment (topisani) [reply server/package.json]: Honestly, i don't know a lot about typescript, and it seemed like i needed to do this. Ill remove them if it works fine without (which i assume it does)

File: server/src/completion.ts (hunk 3 above)
Comment (topisani) [reply server/src/completion.ts]: Oh, so sorry about those - ill fix up these things, and push my changes

File: server/src/completion.ts (hunk 2 above)
Comment (topisani) [reply server/src/completion.ts]: To make relative includes work
